### PR TITLE
Provides variants for methods using the lang abbreviation

### PR DESCRIPTION
**Pull request by sabieber**
Writing language instead of lang is a lot more readable and faster to understand. Its also more consistent with other existing methods.

### CODE REVIEW

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 1]
```diff
@@ -423,9 +471,7 @@ public void setLang(@Nullable String lang) {
      * @param lang the two-letter language code for this thread.
      */
     public void setLangIfEmpty(@Nullable String lang) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java]: deprecate?

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 2]
```diff
@@ -442,27 +488,54 @@ public void setLangIfEmpty(@Nullable String lang) {
...
+     * its session or it might contain a language header which itself isn't quite easy to parse.
+     * <p>
+     * Worst of all, in many cases, the current language might not be used at all.
+     * <p>
+     * Therefore, we permit lazy computations which are only evaluated as required.
+     *
+     * @param languageInstaller a callback which installs the appropriate language into the given call context
+     *                          (is passed again to support {@link #fork() forking}).
+     */
+    public void deferredSetLang(@Nonnull Consumer<CallContext> languageInstaller) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:508]: deprecate?

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 3]
```diff
@@ -442,27 +488,54 @@ public void setLangIfEmpty(@Nullable String lang) {
...
+        language = null;
+    }
+
     /**
      * Sets the lang back to <tt>null</tt>.
      * <p>
-     * This method should only be used to re-initialize the language. Use {@link #setLang(String)} to specify a new language.
+     * This method should only be used to re-initialize the language. Use {@link #setLanguage(String)} to specify a new language.
      */
     public void resetLang() {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:526]: deprecate?

File: src/main/java/sirius/kernel/nls/NLS.java [hunk 4]
```diff
@@ -110,11 +110,22 @@ private NLS() {
...
+    }
+
+    /**
+     * Returns the currently active language as two-letter code.
+     *
+     * @return a two-letter code of the currently active language, as defined in
+     * {@link sirius.kernel.async.CallContext#getLanguage()}
      */
     @Nonnull
     public static String getCurrentLang() {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/nls/NLS.java:127]: Deprecate?

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 5]
```diff
@@ -442,27 +488,54 @@ public void setLangIfEmpty(@Nullable String lang) {
...
      * Sets the current fallback language for the current thread.
      *
-     * @param fallbackLang the two-letter language code for this thread.
+     * @param fallbackLanguage the two-letter language code for this thread.
      */
-    public void setFallbackLang(@Nullable String fallbackLang) {
-        this.fallbackLang = fallbackLang;
+    public void setFallbackLanguage(@Nullable String fallbackLanguage) {
+        this.fallbackLanguage = fallbackLanguage;
     }
```
Comment (cschierle) [inline src/main/java/sirius/kernel/async/CallContext.java:540]: No legacy variant for `setFallbackLang`?

**Review by cschierle — APPROVED** [review]

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 6]
```diff
@@ -405,15 +424,44 @@ public String getFallbackLang() {
...
+     * @param language the two-letter language code for this thread.
      */
-    public void setLang(@Nullable String lang) {
-        if (Strings.isFilled(lang)) {
-            this.lang = lang;
+    public void setLang(@Nullable String language) {
+        if (Strings.isFilled(language)) {
+            this.language = language;
             this.lazyLanguageInstaller = null;
         }
```
[the remark this answers is not in the document]
Comment (cschierle) [reply src/main/java/sirius/kernel/async/CallContext.java]: Whoops, sorry the return is wrong 😅  Should just be 
```
        setLanguage(language);

**Review by andyHa — APPROVED** [review]

**Pull request merged**